t/comp/proto.t

```diff
@@ -892,3 +892,16 @@ $_ = sub ($$$$$$$) {};
 @_ = (1, 2, 3, prototype(), 4, 5, 6);
 print "not " unless "@_" eq '1 2 3 $$$$$$$ 4 5 6';
 print "ok ", $i++, " - [perl #123514] (got @_)\n";
+
+{
+    $i++;
+    my $err = q<>;
+
+    eval 'sub scalarref (\$) { }; scalarref( sub {} )';
+    if ($@ !~ m/anonymous subroutine/ || $@ !~ m/scalarref/) {
+        $err = $@;
+        warn;
```

Comment:
personally id feel better if this was `print "$_\n" for split /\n/, $@;` or `diag` if its available.

FWIW, if I had written this it would look more like this (untested):
```
my $unexpected_err = "";
my $ok = eval 'sub scalarref (\$) { }; scalarref( sub {} ); 1';
my $err = !$ok ? $@ : "";
if ($err !~ m/anonymous subroutine/ || $err !~ m/scalarref/) {
   $unexpected_err = $err;
}
print( (($ok || $unexpected_err) ? 'not' : '') . "ok $i - anonsub passed to \\\$\n",
   map { "#$_\n" } split /\n/, $unexpected_err );
```
I dont mean to style nit, its probably fine as is, but it makes me leery using $@ in any way other than a plain copy. It also makes me leery not checking that eval returns true, and outputing the error on STDERR when the test output is on STDOUT means that it might be emitted out of order which makes life difficult. Using a single print statement with commented output means it should be output all together in one chunk in the correct order and not upset the harness.  The other changes are just to make intent clear here. When I first read this it was obvious what '$err' "meant" in the code. 

None of this is absolutely necessary to change I suspect. But it would make me feel better if you integrated some of the points I raise here.

Reply:
> It also makes me leery not checking that eval returns true

Does checking `$@` not substitute?

Reply:
Tweaked, @demerphq. Thanks!

Reply:
Historically you couldn't rely on $@ being true when an eval fails. The only reliable way to tell that you should look at $@ is if the eval returned a true value.  Maybe this has been fixed enough my concern is irrelevant, but at the same time, using the robust formulation is obviously correct, whereas not using it open up that tiny wedge of doubt. So i just always use the robust formulation, then i don't have to worry. :-)